crates/ruff_linter/src/settings/mod.rs

```diff
@@ -38,13 +41,34 @@ pub mod flags;
 pub mod rule_table;
 pub mod types;
 
+#[macro_export]
+macro_rules! display_settings {
+    (formatter = $fmt:ident, namespace = $namespace:literal, fields = [$($settings:ident.$field:ident $(| $modifier:tt)?),* $(,)?]) => {
+        {
+            const _NS: &str = $namespace;
+            $(
+                display_settings!(@field $fmt, _NS, $settings.$field $(| $modifier)?);
+            )*
+        }
+    };
+    (@field $fmt:ident, $namespace:ident, $settings:ident.$field:ident | debug) => {
+        writeln!($fmt, "{}{} = {:?}", $namespace, stringify!($field), $settings.$field)?;
+    };
+    (@field $fmt:ident, $namespace:ident, $settings:ident.$field:ident | nested) => {
+        write!($fmt, "{}", $settings.$field)?;
+    };
+    (@field $fmt:ident, $namespace:ident, $settings:ident.$field:ident) => {
+        writeln!($fmt, "{}{} = {}", $namespace, stringify!($field), $settings.$field)?;
+    };
+}
```

Comment:
Oh this isn't nearly as bad as I thought it was going to be based on your comments in Discord. :P I'm totally cool with this.

Reply:
That's probably because you're seeing the revised version :P

Thanks though, I'm glad this is workable and understandable as a solution!